### PR TITLE
 Resolving empty package printing a package in source gen if there's no package declaration

### PR DESCRIPTION
**Pull request by NatashaWso2**

### CODE REVIEW

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx [hunk 1]
```diff
@@ -88,7 +88,24 @@ class PackageDeclarationNode extends React.Component {
                 packageNameEditing: false,
             });
         }
-        this.props.model.setPackageName(this.state.packageDefValue);
+        if (this.state.packageDefValue) {
+            const pkgName = 'package ' + this.state.packageDefValue + ';';
```
Comment (hemikak) [inline modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx]: Best to use string templates here.

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx [hunk 2]
```diff
@@ -109,6 +126,19 @@ class PackageDeclarationNode extends React.Component {
             this.onPackageInputBlur(event);
         }
     }
+
+    /**
+     * Get the full package name
+     */
+    getPackageName(model) {
```
Comment (hemikak) [inline modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx]: Return value is missing from the documentation.

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx [hunk 3]
```diff
@@ -199,8 +215,18 @@ class PackageDeclarationNode extends React.Component {
      */
     render() {
         const model = this.props.model;
-        const bBox = model.viewState.bBox;
+        // TODO fix for the model not being parsed properly
+        if (model.viewState.components.topLevelNodes === undefined) {
+            return <g />;
```
Comment (hemikak) [inline modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx]: If you want to return an empty value, you can use
`return (null);`

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx [hunk 4]
```diff
@@ -199,8 +215,18 @@ class PackageDeclarationNode extends React.Component {
...
+        }
+        const bBox = model.viewState.components.topLevelNodes;
         const headerHeight = packageDefinition.header.height;
+
+        // TODO fix for the model not being parsed properly
+        if (this.props.model.viewState.components.importDeclaration === undefined ||
+            this.props.model.viewState.components.importsExpanded === undefined) {
+            return <g />;
```
Comment (hemikak) [inline modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx]: If you want to return an empty value, you can use
`return (null);`

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx (hunk 1 above)
Comment (NatashaWso2) [reply modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx]: Fixed with commit 03ae4b262b1b3d7bb6b90f4060106e4e4fae1334

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx (hunk 2 above)
Comment (NatashaWso2) [reply modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx]: Fixed with commit 03ae4b2

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx (hunk 3 above)
Comment (NatashaWso2) [reply modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx]: Fixed with commit 03ae4b2

File: modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx (hunk 4 above)
Comment (NatashaWso2) [reply modules/web/js/ballerina/diagram2/views/default/components/nodes/top-level-nodes.jsx]: Fixed with commit 03ae4b2

**Pull request merged**